lang/python/pillow/patches/0001-setup.py-use-patch-from-Debian-and-tweak-it.patch

```diff
@@ -0,0 +1,316 @@
+         if struct.calcsize('l') == 4:
+             machine = os.uname()[4] + '-32'
+         else:
+@@ -98,7 +101,7 @@ def _find_library_dirs_ldconfig():
+         env['LC_ALL'] = 'C'
+         env['LANG'] = 'C'
+ 
+-    elif sys.platform.startswith("freebsd"):
++    elif host_platform.startswith("freebsd"):
+         args = ['/sbin/ldconfig', '-r']
+         expr = r'.* => (.*)'
+         env = {}
+@@ -180,12 +183,12 @@ def get_version():
+ 
+ NAME = 'Pillow'
+ PILLOW_VERSION = get_version()
+-JPEG_ROOT = None
++JPEG_ROOT = os.path.join(os.environ['STAGING_DIR'], "/usr/lib")
+ JPEG2K_ROOT = None
+-ZLIB_ROOT = None
++ZLIB_ROOT = os.path.join(os.environ['STAGING_DIR'], "/usr/lib")
+ IMAGEQUANT_ROOT = None
+-TIFF_ROOT = None
+-FREETYPE_ROOT = None
++TIFF_ROOT = os.path.join(os.environ['STAGING_DIR'], "/usr/lib")
++FREETYPE_ROOT = os.path.join(os.environ['STAGING_DIR'], "/usr/lib")
+ LCMS_ROOT = None
+ 
+ 
+@@ -284,6 +287,38 @@ class pil_build_ext(build_ext):
+                 _dbg('Requiring %s', x)
+                 self.feature.required.add(x)
+ 
++    def add_gcc_paths(self):
++        gcc = sysconfig.get_config_var('CC')
++        tmpfile = os.path.join(self.build_temp, 'gccpaths')
++        if not os.path.exists(self.build_temp):
++            os.makedirs(self.build_temp)
++        ret = os.system('%s -E -v - </dev/null 2>%s 1>/dev/null' % (gcc, tmpfile))
++        is_gcc = False
++        in_incdirs = False
++        inc_dirs = []
++        lib_dirs = []
++        try:
++            if ret >> 8 == 0:
++                with open(tmpfile) as fp:
++                    for line in fp.readlines():
++                        if line.startswith("gcc version"):
++                            is_gcc = True
++                        elif line.startswith("#include <...>"):
++                            in_incdirs = True
++                        elif line.startswith("End of search list"):
++                            in_incdirs = False
++                        elif is_gcc and line.startswith("LIBRARY_PATH"):
++                            for d in line.strip().split("=")[1].split(":"):
++                                d = os.path.normpath(d)
++                                if '/gcc/' not in d:
++                                    _add_directory(self.compiler.library_dirs,
++                                                   d)
++                        elif is_gcc and in_incdirs and '/gcc/' not in line:
++                            _add_directory(self.compiler.include_dirs,
++                                           line.strip())
++        finally:
++            os.unlink(tmpfile)
++
+     def build_extensions(self):
+ 
+         library_dirs = []
+@@ -349,115 +384,12 @@ class pil_build_ext(build_ext):
+                 for d in os.environ[k].split(os.path.pathsep):
+                     _add_directory(library_dirs, d)
+ 
+-        prefix = sysconfig.get_config_var("prefix")
++        prefix = os.environ.get('PYTHONXCPREFIX')
```

Comment:
curios: is this necessary ?
typically the prefix can be obtained from the CONFIGURE_ARGS reasonably ok
if this isn't true, we can try to fix it
i wouldn't mind having this here either; but if we could not patch this, it would be better

Reply:
Also, it already [reads from several environment variables](https://github.com/python-pillow/Pillow/blob/6.0.0/setup.py#L334-L350), which should have all the paths it needs?

Also... I don't see where `PYTHONXCPREFIX` is defined.

I would be interested to see what `sysconfig.get_config_var("prefix")` returns.